src/Microsoft.Framework.Runtime.Hosting/Json/JsonObject.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Framework.Runtime.Json
+{
+    internal class JsonObject : JsonValue
+    {
+        private readonly IDictionary<string, JsonValue> _data;
+
+        public JsonObject(IDictionary<string, JsonValue> data, int line, int column)
+            : base(line, column)
+        {
+            if (data == null)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            _data = data;
+        }
+
+        public ICollection<string> Keys
+        {
+            get { return _data.Keys; }
+        }
+
+        public JsonValue Value(string key)
+        {
+            JsonValue result;
+            if (!_data.TryGetValue(key, out result))
+            {
+                result = null;
+            }
+
+            return result;
+        }
+
+        public T ValueAs<T>(string key, Func<JsonValue, T> cast)
```

Comment:
This method doesn't seem terribly useful. Why can't the caller just grab the `JsonValue` and do whatever it wants? A method that takes in a delegate and just calls that delegate seems very roundabout.